opensquirrel/squirrel_ir.py

```diff
@@ -77,10 +83,59 @@ def accept(self, visitor: SquirrelIRVisitor):
         return visitor.visit_qubit(self)
 
 
+@dataclass
+class Bit(Expression):
+    index: int
+
+    def __hash__(self):
+        return hash(self.index)
+
+    def __repr__(self):
+        return f"Bit[{self.index}]"
+
+    def accept(self, visitor: SquirrelIRVisitor):
+        return visitor.visit_bit(self)
+
+
```

Comment:
This is only required if we don't deprecate the antlr parser, which we will, so it's not needed.